### DISCUSSION TITLE
Sonoff ZbBridge with Hue Bridge emulation, emulates old Firmware

### DISCUSSION THREAD

**Discussion by Barracuda09** [arendst/Tasmota#10940] [Ideas]
When opening the "_Philips Hue Bridge v1_"  android app, it will try to update the firmware. So the _Hue emulation_, emulates older firmware.

And maybe therefore Alexa will not discover the Hue bridge. (**Edit: This is not the cause of the discovery problem**)

Tasmota binary firmware version number used: **9.2.0.4(zbbridge)**

### EXPECTED BEHAVIOUR
That the app will not try to update firmware

**Comment by Barracuda09** [comment]
Well this will probably not be resolved here. It is a bug in my opinion, so no real discussion issue.

**Comment by Jason2866** [comment]
No, it is not a bug. It is as designed. Discovering devices and using Tasmota devices with Alexa does work.
You cant update the HueBridge since it is not a real one. It is a Software Emulation.
Using HueEmulation in a ZbBridge makes no sense at all. There is no function supported.
Solution. Just switch off the HueEmulation in the ZbBridge

**Reply by bwims** [reply]
If you turn off the HueEmulation in the ZbBridge, how will Alexa discover new non-Philips Zigbee lights?

Incidentally mine was working fine with discovered lights until about a week ago, when I needed to change its IP address, as reported [here](https://github.com/arendst/Tasmota/issues/11526)

**Reply by Barracuda09** [reply]
It is not an anwer to my question. @ascillato keeps marking this as answer.

**Reply by s-hadinger** [reply]
It is already answered below. New Hue protocol requires HTTPS which is beyond ESP8266 reach. Unless the V1 is broken, there is no need to update to V2.

**Reply by Barracuda09** [reply]
No it is not V2. Read the description

> "_Philips Hue Bridge v1_" android app

**Reply by ascillato** [reply]
Please, update your Tasmota to latest version http://ota.tasmota.com/tasmota/ 

Thanks.

**Reply by Barracuda09** [reply]
@ascillato Please stop marking this as answered with a bogus answer.

The latest version will most likely not fix this, as I know what the problem is.

**Reply by bwims** [reply]
I'm running 9.3.1.2 on my zbbridge and dicovery yields zilch.

**Reply by Barracuda09** [reply]
@bwims 

This is not (anymore) about the discovery problem. As I started out this Alexa issue some time ago, I had the idea it was the problem. But it turned out to be something else that was fixed. But this issue is still not really fixed here. I have done some changes that will make the  _"Philips Hue Bridge v1"_ android app work again. But there are some more issues I have to checkout as it just keeps on crashing the zbbridge with an exception when I try to define rooms. But in time I will fix this as well.

That is way I insist that this is not answered.

**Reply by s-hadinger** [reply]
It looks to me the original question is answered. Tasmota will not move to Hue v2. Please don't mix different problems in the same thread.

**Reply by Barracuda09** [reply]
Please this is not about Hue V2. This is about V1 and **Philips Hue Bridge v1** app see description

But have it your way. I am done with this.

**Reply by s-hadinger** [reply]
We don't support Hue app.

**Reply by Barracuda09** [reply]
Clearly you don't mean this?

**Reply by Barracuda09** [reply]
Its not more then just Alexa its the same API

**Reply by Barracuda09** [reply]
Only checks are stricter

**Reply by ascillato** [reply]
> Its not more then just Alexa its the same API. Only checks are stricter.

Cool. If you could help us with a PR would be awesome.

**Reply by ascillato** [reply]
Or with the required checks you mention.

**Reply by Barracuda09** [reply]
Yes, I could do that. But as I mentioned before there are some crashes in current API implementation.

But sorry, atmosphere here is not nice. You keep marking this as answered.

**Reply by ascillato** [reply]
Hi, this discussion has been moved from question to idea. Idea seems to be better for this case as it is about a feature enhance discussion more than just a question and answer. So far,  Ideas can not be marked in github as answered or not.

Do you agree with this category change?

**Comment by Barracuda09** [comment]
Well you can argue it is by design, than I say it is maybe an outdated design.

**Reply by ascillato** [reply]
If you want, you can modify the driver to emulate a newer version. A PR is always welcome due to Tasmota is made from contributions from the community.

**Reply by Barracuda09** [reply]
I will give it a try in the near future.

**Reply by s-hadinger** [reply]
I bzliece that newer protocols use https, which makes it not possible on esp8266.

**Reply by Barracuda09** [reply]
I don't think so, its Hue bridge V1
